### PR TITLE
Added Redux and Roles

### PR DESCRIPTION
**Pull request by moizahmed102**
Added Redux toolkit for user state management and roles for user and admin.

### CODE REVIEW

File: backend/controllers/taskController.js [hunk 1]
```diff
@@ -5,20 +5,37 @@ const getTasks = async (req, res) => {
...
-    res.status(200).json({
-      status: "Success",
-      data: {
-        totalTasks,
-        paginatedTasks,
-      },
-    });
+    let tasks;
+
+    if (req.user.role === "admin") {
```
Comment (numan-ahmad) [inline backend/controllers/taskController.js]: please use one query for both

File: frontend/src/components/Login.jsx [hunk 2]
```diff
@@ -33,14 +26,21 @@ function Login() {
       [event.target.name]: event.target.value,
     }));
   };
-
+  useEffect(() => {
+    console.log("User:", user);
```
Comment (numan-ahmad) [inline frontend/src/components/Login.jsx]: remove console

**Review by numan-ahmad — COMMENTED** [review]
Please use the intercepter for handle the error globally

File: backend/controllers/userController.js [hunk 3]
```diff
@@ -1,21 +1,22 @@
 const jwt = require("jsonwebtoken");
 const bcrypt = require("bcryptjs");
 const User = require("../models/userModel");
-const Createtodo = require("../models/taskModel");
 
-const token = (id) => {
-  return jwt.sign({ id }, process.env.JWT_KEY, { expiresIn: "20d" });
+const token = (user) => {
+  return jwt.sign({ id: user._id, role: user.role }, process.env.JWT_KEY, {
+    expiresIn: "20d",
```
Comment (numan-ahmad) [inline backend/controllers/userController.js]: please set this env and good practice is to set expireIn only 1 day

File: backend/models/taskModel.js [hunk 4]
```diff
@@ -1,15 +1,18 @@
...
-  user: {
-    type: mongoose.Schema.Types.ObjectId,
-    ref: "User",
-    required: true,
-  },
-});
+  { timestamps: true }
+);
 
 module.exports = mongoose.model("Createtodo", taskSchema);
```
Comment (numan-ahmad) [inline backend/models/taskModel.js]: please change the name this is schema so set name like Todo not createTodo

File: frontend/src/components/AdminDashboard.jsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    fetchTasks(); 
+  }, [currentPage, navigate]);
+
+  const fetchTasks = async () => {
+    try {
+      const data = await getAllTasksForAdmin(currentPage);
+      setTasks(data.data.tasks || []);
+      setTotalTasks(data.data.totalTasks || 0);
+    } catch (error) {
+      console.error(error);
```
Comment (numan-ahmad) [inline frontend/src/components/AdminDashboard.jsx]: remove console

File: frontend/src/components/AdminDashboard.jsx [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+  };
+
+  const handleUpdateTask = async (taskId) => {
+    try {
+      await updateTask(taskId, { task: newTaskText });
+      setEditingTaskId(null);
+      setNewTaskText('');
+      fetchTasks(); 
+    } catch (error) {
+      console.log("Error updating task:", error);
```
Comment (numan-ahmad) [inline frontend/src/components/AdminDashboard.jsx]: please use tost library for show error or success

**Review by numan-ahmad — COMMENTED** [review]
please remove the package-lock file

File: backend/controllers/taskController.js (hunk 1 above)
Comment (moizahmed102) [reply backend/controllers/taskController.js]: > please use one query for both

resolved.

File: frontend/src/components/Login.jsx (hunk 2 above)
Comment (moizahmed102) [reply frontend/src/components/Login.jsx]: > remove console

resolved

**Comment by moizahmed102** [comment]
> Please use the intercepter for handle the error globally

resolved

File: backend/controllers/userController.js (hunk 3 above)
Comment (moizahmed102) [reply backend/controllers/userController.js]: resolved.

File: backend/models/taskModel.js (hunk 4 above)
Comment (moizahmed102) [reply backend/models/taskModel.js]: resolved.

File: frontend/src/components/AdminDashboard.jsx (hunk 5 above)
Comment (moizahmed102) [reply frontend/src/components/AdminDashboard.jsx]: resolved.

File: frontend/src/components/AdminDashboard.jsx (hunk 6 above)
Comment (moizahmed102) [reply frontend/src/components/AdminDashboard.jsx]: resolved.

**Comment by moizahmed102** [comment]
> please remove the package-lock file

resolved.

**Pull request merged**